### DISCUSSION TITLE
Issue regarding rotation of foundry blackboxes

### DISCUSSION THREAD

**Discussion by manoranjanminz** [gdsfactory/gdsfactory#2191] [Q&A]
Hi,

I am working with foundry blackboxes (BBs), and for a design I need to rotate it by 45 degrees. When the BBs are not rotated, the pins of the BBs are represented by boxes. But the problem that I am facing right now is that after rotation, the pins get converted to polygons and they snap to the grid, and they slightly come out of the BB layer. 

![image](https://github.com/gdsfactory/gdsfactory/assets/102401390/e8bbf22a-92c5-43fa-8151-22eb97f4f3d6)

The reason I believe this is because the rotate command flattens the instances after rotation. I could reproduce the issue in KLayout when I rotated and flattened the BB.

Is there a way I could resolve this?

**Comment by joamatab** [comment]
yes, that's quite tricky

that's the reason i never rotate references at a non-manhattan angle (0,90, 180, 270)

**Comment by joamatab** [comment]
we added a tutorial for this https://gdsfactory.github.io/gdsfactory/notebooks/04_routing_non_manhattan.html

**Comment by joamatab** [comment]
This is related to https://github.com/gdsfactory/gdsfactory/issues/2506


that's one of the limitations of black boxes, here are some options I see:

- avoid non-manhattan rotations
- don't use black boxes
- don't and have `flatten_offgrid_references`  and live with 1nm gaps or ask Ligentec to run `flatten_offgrid_references` on their end

**Reply by manoranjanminz** [reply]
Thanks @joamatab ,

I appreciate your continuous effort towards resolution of this particular issue. 

I guess I will have to resort to the last option that you mentioned. I need to communicate with the foundry regarding this. Please keep me posted on this thread if a better solution comes up.

Regards,
Manoranjan
